core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -52,10 +52,37 @@
 #include <HIP/Kokkos_HIP_Instance.hpp>
 #include <Kokkos_HIP.hpp>
 
+// TODO cannot use global variable on the device with ROCm 2.9
+//__device__ __constant__ unsigned long kokkos_impl_hip_constant_memory_buffer
+//    [Kokkos::Experimental::Impl::HIPTraits::ConstantMemoryUsage /
+//     sizeof(unsigned long)];
+
+namespace Kokkos {
+namespace Experimental {
+template <typename T>
+inline __device__ T *kokkos_impl_hip_shared_memory() {
+  extern __shared__ HIPSpace::size_type sh[];
```

Comment:
You know about the `HIP_DYNAMIC_SHARED` macro?  (I don't expect you to change)

Reply:
No, I just saw that `__shared__` worked like in CUDA, so I just used it.